plugins/httpsreq/httpsreqfunc.cpp

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2022 Second State INC
+
+#include "httpsreqfunc.h"
+#include <errno.h>
+#include <iostream>
+#include <netdb.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <resolv.h>
+#include <string.h>
+#include <unistd.h>
+
+// Some of the code was taken from this post:
+// https://stackoverflow.com/questions/52727565/client-in-c-use-gethostbyname-or-getaddrinfo
+
+namespace WasmEdge {
+namespace Host {
+
+Expect<void> SendData::body(Runtime::Instance::MemoryInstance *MemInst,
+                            uint32_t HostPtr, uint32_t HostLen, uint32_t Port,
+                            uint32_t BodyPtr, uint32_t BodyLen) {
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::ExecutionFailed);
+  }
+
+  char *Host = MemInst->getPointer<char *>(HostPtr);
+  std::string NewHost;
+  std::copy_n(Host, HostLen, std::back_inserter(NewHost));
+  Env.Host = std::move(NewHost);
+
+  Env.Port = Port;
+
+  char *BodyStr = MemInst->getPointer<char *>(BodyPtr);
+  std::string NewBodyStr;
+  std::copy_n(BodyStr, BodyLen, std::back_inserter(NewBodyStr));
+  Env.BodyStr = std::move(NewBodyStr);
+
+  const SSL_METHOD *Method = TLS_client_method();
+  SSL_CTX *Ctx = SSL_CTX_new(Method);
+
+  if (Ctx == nullptr) {
+    ERR_print_errors_fp(stderr);
+    exit(EXIT_FAILURE);
+  }
+
+  SSL *Ssl = SSL_new(Ctx);
+  if (Ssl == nullptr) {
+    fprintf(stderr, "SSL_new() failed\n");
```

Comment:
If you want to use the `stderr` directly, please add a prefix string which is the name of your tool. Otherwise, you should use our log system to print it.

Reply:
ok